spec/features/movies/show_spec.rb

```diff
@@ -0,0 +1,105 @@
+      frank = ddarko.actors.create!(name: "Frank", age: 18)
+
+      visit "/movies/#{ddarko.id}"
+
+      expect(frank.name).to appear_before(jake.name)
+      expect(jake.name).to appear_before(maggie.name)
+      expect(maggie.name).to_not appear_before(frank.name)
+
+      wb = Studio.create!(name: "Warner Brothers", location: "Burbank")
+      jodirt = wb.movies.create!(title: "Joe Dirt", creation_year: "2000", genre: "Comedy")
+      spade = jodirt.actors.create!(name: "David Spade", age: 46)  
+      walken = jodirt.actors.create!(name: "Christopher Walken", age: 74)  
+
+      visit "/movies/#{jodirt.id}"
+
+      expect(spade.name).to appear_before(walken.name)
+    end
+
+    it "has average age of all movies actors" do 
+      uni = Studio.create!(name: "Universal Studios", location: "Hollywood")
+      ddarko = uni.movies.create!(title: "Donnie Darko", creation_year: "2000", genre: "Sci-Fi")
+      jake = ddarko.actors.create!(name: "Jake Gyllenhaal", age: 42)   
+      maggie = ddarko.actors.create!(name: "Maggie Gyllenhaal", age: 45)
+      frank = ddarko.actors.create!(name: "Frank", age: 18)
+
+      visit "/movies/#{ddarko.id}"
+
+      expect(page).to have_content("Average Age of Actors: #{(jake.age + maggie.age + frank.age)/ddarko.actors.count}")
+
+      wb = Studio.create!(name: "Warner Brothers", location: "Burbank")
+      jodirt = wb.movies.create!(title: "Joe Dirt", creation_year: "2000", genre: "Comedy")
+      spade = jodirt.actors.create!(name: "David Spade", age: 46)  
+      walken = jodirt.actors.create!(name: "Christopher Walken", age: 74) 
+
+      visit "/movies/#{jodirt.id}"
+
+      expect(page).to have_content("Average Age of Actors: #{(spade.age + walken.age)/jodirt.actors.count}")
+    end
+
+    it "does not list actors that are not in the movie" do 
+      wb = Studio.create!(name: "Warner Brothers", location: "Burbank")
+      jodirt = wb.movies.create!(title: "Joe Dirt", creation_year: "2000", genre: "Comedy")
+      spade = jodirt.actors.create!(name: "David Spade", age: 46)  
+      walken = jodirt.actors.create!(name: "Christopher Walken", age: 74) 
+      robert = Actor.create!(name: "Bobby Dinero", age: 74) 
+
+
+      visit "/movies/#{jodirt.id}"
+
+      expect(page).to_not have_content(robert.name)
+    end
+
+    it "has a form to add actor to movie" do 
+      wb = Studio.create!(name: "Warner Brothers", location: "Burbank")
+      jodirt = wb.movies.create!(title: "Joe Dirt", creation_year: "2000", genre: "Comedy")
+      spade = jodirt.actors.create!(name: "David Spade", age: 46)  
+      walken = jodirt.actors.create!(name: "Christopher Walken", age: 74) 
+      rock = Actor.create!(name: "Kid Rock", age: 52)
+      visit "/movies/#{jodirt.id}"
+
+      fill_in("Actor ID", with: rock.id)
+      click_button "Submit"
+      expect(current_path).to eq("/movies/#{jodirt.id}")
+      expect(page).to have_content("Kid Rock")
+    end
+  end
+end
```

Comment:
Your tests are well written, but make sure you're using them to guide in your development and read test failures completely to get a better sense of what could be going wrong. It can be daunting to look through stack traces but it's a super important skill and will serve you really well!